internal/controller/config.go

```diff
@@ -0,0 +1,42 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/spf13/viper"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// GlobalConfig is the configuration for the controllers, it is
+// supposed to be populated from the configmap attached to the
+// controller manager.
+type GlobalConfig struct {
+	ClusterName     string                               `yaml:"clusterName" json:"clusterName"`
+	TimeInterval    uint                                 `yaml:"timeInterval" json:"timeInterval"`
+	RepoCredentials map[string]*corev1.SecretKeySelector `json:"repoCredentials"`
```

Comment:
I'm kind of thinking we should have a new struct called `GitCredential` and map to that. The `GitCredential` struct can consist of:
```
type GitCredential struct {
    httpCredential *HTTPCredential 
    sshCredential *SSHCredential
}

type HTTPCredential struct {
    Username string
    Password string
}
type SSHCredential struct {
   Name string
   SSHKey string
}
```

Let me get back to you on also incorporating TLS settings into this...